### DISCUSSION TITLE
How to creating a janusgraph cluster ?

### DISCUSSION THREAD

**Discussion by TouSS** [JanusGraph/janusgraph#2907] [Q&A] [answered]
Guys,

I'm currently struggling in creating a Janusgraph cluster. I read the official documents and i'm not clear yet how a cluster is created.

My problem is how can we create a janusgraph/gremlin server cluster?
     * Do i only need to run janusgraph/gremlin servers in the same network so that they will create a cluster automatically? 
     * Or is there a way to configure a cluster so that janusgraph instances will communicate with each other?
     * If there is no communication between the janusgraph/gremlin servers can't there be any transaction issues?  For example one server adding a new node to a graph and at the same time another server is also adding a node which  will make an inconsistency.
    
OR, Is right that janusgraph read the data from backed storage, Then know each node of the cluster ??

Thanks and regards

**Comment by li-boxuan** [comment, marked as answer]
A JanusGraph cluster is mostly a logical concept. They don't communicate directly. Rather, they communicate via the underlying backend storage. Race conditions can be avoided by using [locking](https://docs.janusgraph.org/advanced-topics/eventual-consistency/), which uses the underlying backend storage as the communication bridge.

**Comment by TouSS** [comment]
[Dynamic Graphs](https://docs.janusgraph.org/operations/dynamic-graphs/), In this section, I found:
> For example, if you update or delete the configuration to a graph on one JanusGraph node, then we must evict that graph from the cache on every JanusGraph node in the cluster. Otherwise, we may have inconsistent graph representations across your cluster. JanusGraph automatically handles this eviction using <b>"a messaging log queue through the backend system"</b> that the graph in question is configured to use.

What about the "a messaging log queue through the backend system", If this "messaging log queue"  must be support by backend storage? OR, It is part of JanusGraph.
If It is part of JanusGraph, this is no communicate for each on, How it work?

**Reply by li-boxuan** [reply]
The messaging queue is very simple. Nodes write to the backend storage and/or read from the backend storage. The backend storage is used as a "broker". No special requirement is needed for the backend storage.

**Reply by TouSS** [reply]
Got it !
